packages/jest-matchers/src/__tests__/toThrowMatchers-test.js

```diff
@@ -116,24 +117,44 @@ class Error {
       });
 
       test('did not throw at all', () => {
-        matchErrorSnapshot(() => expect(() => {})[toThrow](Err));
+        expect(() => expect(() => {})[toThrow](Err))
+          .toThowErrorMatchingSnapshot();
       });
 
       test('threw, but class did not match', () => {
-        matchErrorSnapshot(() => {
+        expect(() => {
           jestExpect(() => { throw new Err('apple'); })[toThrow](Err2);
-        });
+        }).toThowErrorMatchingSnapshot();
       });
 
       test('threw, but should not have', () => {
-        matchErrorSnapshot(() => {
+        expect(() => {
           jestExpect(() => { throw new Err('apple'); }).not[toThrow](Err);
-        });
+        }).toThowErrorMatchingSnapshot();
       });
     });
 
     test('invalid arguments', () => {
-      matchErrorSnapshot(() => jestExpect(() => {})[toThrow](111));
+      expect(() => jestExpect(() => {})[toThrow](111))
+        .toThowErrorMatchingSnapshot();
+    });
+  });
+
+
+  describe('.toThowErrorMatchingSnapshot()', () => {
+    it('does not accept arguments', () => {
+      expect(() => {
+        jestExpect(() => { throw new Error('apple'); })
+          .toThowErrorMatchingSnapshot('foobar');
+      }).toThowErrorMatchingSnapshot();
+    });
+
+    it("throws the error if tested function didn't throw error", () => {
+      expect(() => {
+        jestExpect(() => {}).toThowErrorMatchingSnapshot();
+      }).toThrowError(
```

Comment:
this one messed up tests though... if i do it like

```javascript
    it("throws the error if tested function didn't throw error", () => {
      expect(() => {
        jestExpect(() => {}).toThowErrorMatchingSnapshot();
      }).toThowErrorMatchingSnapshot();
    });
```
it work fine on first run while it generating snapshot but on second run getting this error

```
  ● .toThowErrorMatchingSnapshot() › throws the error if tested function didn't throw error

    Received value does not match the stored snapshot 1.

    null

      at Object.<anonymous> (packages/jest-matchers/src/__tests__/toThrowMatchers-test.js:155:7)


Snapshot Summary
 › 1 snapshot written in 1 test file.
 › 1 snapshot test failed in 1 test file. Inspect your code changes or run with `npm run jest -- -u` to update them.
```